packages/phone-number-privacy/combiner/src/common/handlers.ts

```diff
@@ -18,31 +27,62 @@ export type PromiseHandler<R extends OdisRequest> = (
   res: Response<OdisResponse<R>, Locals>
 ) => Promise<void>
 
-type ParentHandler = (req: Request<{}, {}, any>, res: Response<any, Locals>) => Promise<void>
-
 export function catchErrorHandler<R extends OdisRequest>(
   handler: PromiseHandler<R>
-): ParentHandler {
+): PromiseHandler<R> {
   return async (req, res) => {
-    const logger: Logger = res.locals.logger
     try {
       await handler(req, res)
     } catch (err) {
+      const logger: Logger = res.locals.logger
       logger.error(ErrorMessage.CAUGHT_ERROR_IN_ENDPOINT_HANDLER)
       logger.error(err)
       if (!res.headersSent) {
-        logger.info('Responding with error in outer endpoint handler')
-        res.status(500).json({
-          success: false,
-          error: ErrorMessage.UNKNOWN_ERROR,
-        })
+        if (err instanceof OdisError) {
+          sendFailure(err.code, err.status, res, req.url)
+        } else {
+          sendFailure(ErrorMessage.UNKNOWN_ERROR, 500, res, req.url)
+        }
       } else {
         logger.error(ErrorMessage.ERROR_AFTER_RESPONSE_SENT)
       }
     }
   }
 }
 
+export function tracingHandler<R extends OdisRequest>(
```

Comment:
This could be moved to `tracing-utils.ts`

Reply:
same